### PR TITLE
refactor(frontend): make frontend plugin stateless

### PR DESCRIPTION
**Pull request by hund030**
1. Make frontend plugin stateless.
2. Remove useless codes and comments.
3. Sync config to context, make it easier to add new persistent configs.

### CODE REVIEW

File: packages/fx-core/src/plugins/resource/frontend/configs.ts [hunk 1]
```diff
@@ -79,21 +74,38 @@ export class FrontendConfig {
...
       location,
-      resourceNameSuffix,
       storageName,
       credentials
     );
   }
 
+  public syncToPluginContext(ctx: PluginContext): void {
+    Object.entries(this)
+      .filter((kv) => FrontendConfig.persistentConfigList.find((x: string) => x === kv[0]))
```
Comment (henzhang-ms) [inline packages/fx-core/src/plugins/resource/frontend/configs.ts:87]: Nit: could be `FrontendConfig.persistentConfigList.find(kv[0])`

File: packages/fx-core/src/plugins/resource/frontend/configs.ts [hunk 2]
```diff
@@ -79,21 +74,38 @@ export class FrontendConfig {
...
+      .forEach((kv) => {
+        if (kv[1]) {
+          FrontendConfig.setConfigIfNotExists(ctx, kv[0], kv[1]);
+        }
+      });
+  }
+
+  private static persistentConfigList = Object.entries(FrontendConfigInfo).map((kv) => kv[1]);
```
Comment (henzhang-ms) [inline packages/fx-core/src/plugins/resource/frontend/configs.ts:95]: Nit: could be `Object.values(FrontendConfigInfo)`.

**Review by henzhang-ms — APPROVED** [review]

File: packages/fx-core/src/plugins/resource/frontend/configs.ts (hunk 2 above)
Comment (hund030) [reply packages/fx-core/src/plugins/resource/frontend/configs.ts]: Updated.

File: packages/fx-core/src/plugins/resource/frontend/configs.ts (hunk 1 above)
Comment (hund030) [reply packages/fx-core/src/plugins/resource/frontend/configs.ts]: seems `find` only consume a callback, use includes instead.

**Pull request merged**